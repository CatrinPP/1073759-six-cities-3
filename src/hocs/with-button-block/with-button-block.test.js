import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withButtonBlock from './with-button-block.js';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
};

const MockComponentWrapped = withButtonBlock(MockComponent);

it(`withButtonBlock is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      setRating={() => {}}
      checkCommentFilled={() => {}}
      uncheckCommentFilled={() => {}}
      isSubmitButtonBlocked={false}
    />
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
