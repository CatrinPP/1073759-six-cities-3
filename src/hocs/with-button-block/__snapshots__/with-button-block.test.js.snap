// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withButtonBlock is rendered correctly 1`] = `<div />`;
